sdk_contrib/aws_v1_support/src/main/java/io/opentelemetry/sdk/contrib/trace/aws/AwsXRayIdsGenerator.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.contrib.trace.aws;
+
+import com.amazonaws.xray.ThreadLocalStorage;
+import com.amazonaws.xray.entities.TraceID;
+import io.opentelemetry.sdk.trace.IdsGenerator;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.TraceId;
+
+/**
+ * Generates tracing ids compatible with the AWS X-Ray tracing service. In the X-Ray system the
+ * first 32 bits of the trace id are the Unix epoch time in secords. Spans (AWS calls them segments)
+ * submit with trace id timestamps outside of the last 30 days are rejected.
+ *
+ * @see <a
+ *     href="https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html#xray-api-traceids">Generating
+ *     Trace IDs</a>
+ */
+public class AwsXRayIdsGenerator implements IdsGenerator {
+
+  @Override
+  public SpanId generateSpanId() {
+    String awsIdStr = generateId();
+    return SpanId.fromLowerBase16(awsIdStr, 0);
+  }
+
+  @Override
+  public TraceId generateTraceId() {
+    String traceIdStr = new TraceID().toString();
```

Comment:
This seems to me that you always generate the same traceid, please comment and explain if I understand wrong.

Maybe consider a different PR for id generator and ec2 resource.

Reply:
It will be a new trace id each time. This is the constructor code.

public TraceID() {
        this(Instant.now().getEpochSecond());
    }

    public TraceID(long startTime) {
        number = new BigInteger(96, ThreadLocalStorage.getRandom());
        this.startTime = startTime;
    }

Reply:
ahhhh, that is the instance from aws :( can we use the full qualified name (including package) when we use that instance to make the code more readable?

Reply:
Changed